runtime/jvmti/jvmtiClass.c

```diff
@@ -2674,27 +2674,26 @@ jvmtiGetConstantPool_addMethodHandle(jvmtiGcp_translation *translation, UDATA cp
 	case MH_REF_PUTSTATIC:
 		fieldOrMethodCpType = CFR_CONSTANT_Fieldref;
 		break;
-
 	case MH_REF_INVOKEVIRTUAL:
 	case MH_REF_INVOKESTATIC:
 	case MH_REF_INVOKESPECIAL:
 	case MH_REF_NEWINVOKESPECIAL:
 		fieldOrMethodCpType = CFR_CONSTANT_Methodref;
 		break;
-
 	case MH_REF_INVOKEINTERFACE:
 		fieldOrMethodCpType = CFR_CONSTANT_InterfaceMethodref;
 		break;
 	default:
-		Assert_JVMTI_true(0);
-		return JVMTI_ERROR_INTERNAL;
```

Comment:
I don't think this return statement should be removed.

Reply:
why?

Reply:
Neither the compiler, nor someone unfamiliar with the meaning of `Assert_JVMTI_unreachable();` will understand that the following code is not executed (with `fieldOrMethodCpType = 0`). It's a trivial cost to pay for the added readability, and in some (although perhaps not this) situations might allow the compiler to generate better code.

Reply:
Re-added.